src/engine/sparqlExpressions/SparqlExpression.h

```diff
@@ -87,6 +87,12 @@ class SparqlExpression {
   // Check if expression is aggregate
   virtual bool isAggregate() const { return false; }
 
+  // Check if an expression is distinct (only applies to aggregates)
+  virtual bool isDistinct() const {
+    AD_CONTRACT_CHECK(this->isAggregate());
```

Comment:
```suggestion
    AD_CONTRACT_CHECK(isAggregate());
```
(This is not Java:))